source/GpioController.cpp

```diff
@@ -79,3 +86,164 @@ HRESULT BcmGpioControllerClass::_mapController()
 }
 #endif // defined(_M_ARM)
 
+#if defined(_M_IX86) || defined(_M_X64)
```

Comment:
It would be nice to have separate `_M_IX86 || _M_X64` and `_M_ARM` implementation files, and then have a...
```
#if defined(_M_IX86) || defined(_M_X64)
#include X.h
#elif
#include ARM.h
#endif
```

Alternatively, you could use the `#elif` directive to easily identify which code block replaces the others architecture's implementation.